pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -265,24 +265,36 @@ func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
 	if err != nil {
 		return err
 	}
+
 	if err := r.client.CreateOrUpdate(&cm); err != nil {
 		return err
 	}
 	return nil
 }
 
-func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig) automationconfig.AutomationConfig {
+func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig, client mdbClient.Client) (automationconfig.AutomationConfig, error) {
 	domain := getDomain(mdb.ServiceName(), mdb.Namespace, "")
-	return automationconfig.NewBuilder().
+
+	currentAc, err := getCurrentAutomationConfig(client, mdb)
+	if err != nil {
+		return automationconfig.AutomationConfig{}, err
+	}
+	newAc, err := automationconfig.NewBuilder().
 		SetTopology(automationconfig.ReplicaSetTopology).
 		SetName(mdb.Name).
 		SetDomain(domain).
 		SetMembers(mdb.Spec.Members).
+		SetPreviousAutomationConfig(currentAc).
```

Comment:
I don't want to block on this, but maybe we can do the call to getCurrentAutomationConfig directly in the build function: as we always want to increment the version if it's different, the call should be by default and the closest possible to build time

you can do it later/never if you don't think it's important

![](https://media.giphy.com/media/Ngu7v2fXqO9a/giphy.gif)